### PR TITLE
Fix incorrect UnhandledMatchError for match expressions using spaceship operator

### PR DESCRIPTION
**Pull request by rvanvelzen**
See https://github.com/phpstan/phpstan/issues/4451#issuecomment-1149919639

### CODE REVIEW

File: src/Analyser/MutatingScope.php [hunk 1]
```diff
@@ -661,6 +661,11 @@ private function resolveType(Expr $node): Type
 			return new NeverType(true);
 		}
 
+		$exprString = $this->getNodeKey($node);
+		if (isset($this->moreSpecificTypes[$exprString]) && $this->moreSpecificTypes[$exprString]->getCertainty()->yes()) {
+			return $this->moreSpecificTypes[$exprString]->getType();
```
Comment (ondrejmirtes) [inline src/Analyser/MutatingScope.php:666]: It's an obvious solution but it's gonna be slow :(

Comment (rvanvelzen) [reply src/Analyser/MutatingScope.php:666]: I didn't notice a slowdown - running `make phpstan` actually seems slightly faster :sweat_smile:

Comment (ondrejmirtes) [reply src/Analyser/MutatingScope.php:666]: Alright, I'll benchmark it :)

**Comment by ondrejmirtes** [comment]
I profiled it and it's not slower 🎉 I even came up with an idea for a small optimization which I'm gonna try soon. Thank you!

**Pull request merged**